### ISSUE TITLE
Could not load compiled classes for initialization script

### ISSUE THREAD

**Issue by arouel** [gradle/gradle-build-action#91]
We often see the following failure when executing jobs with a `gradle-build-action` in various workflows. What they all have in common when the error occurs is that the id `build_scan_capture_init_6jxid5m3sbemobrv5hpcfrrqr` is printed. Is there any advice to work around or fix this problem? Could it be a broken cache entry?
```
Run gradle/gradle-build-action@v2
Restore Gradle state from cache
Writing init script: /home/runner/work/_temp/build-scan-capture.init.gradle
/home/runner/work/project1/project1/gradlew some-task --no-daemon --init-script /home/runner/work/_temp/build-scan-capture.init.gradle
Downloading https://services.gradle.org/distributions/gradle-7.2-all.zip
..............10%...............20%...............30%...............40%...............50%...............60%...............70%...............80%...............90%...............100%
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.2/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build 

FAILURE: Build failed with an exception.

* What went wrong:
Could not load compiled classes for initialization script '/home/runner/work/_temp/build-scan-capture.init.gradle' from cache.
> build_scan_capture_init_6jxid5m3sbemobrv5hpcfrrqr
```

**Comment by mateuszkwiecinski** [comment]
I started seeing the same across different repositories. 
Here's link to a sample failed run: https://github.com/usefulness/project-starter/runs/3842093922

**Comment by bigdaz** [comment]
@arouel @mateuszkwiecinski @balazsbanyai Thanks for your reports.
It looks like the Gradle User Home cache isn't being restored fully, and that my assumptions about what Gradle would tolerate in this regard may not be correct.

To assist with the diagnosis, can I request that you:
1. Enable [cache debug logging](https://github.com/gradle/gradle-build-action/blob/main/README.md#cache-debugging) for the `gradle-build-action`
2. Add `--stacktrace` to your build invocation so that we can see the full failure details.

Then if you see a similar failure in the future, please link to it here. Thanks!

**Comment by bigdaz** [comment]
This issue has been fixed in `v2.0-beta.5`
